### PR TITLE
WIP time-trail

### PR DESCRIPTION
**Pull request by simongasson**
Added queries
Added slider

### CODE REVIEW

**Comment by Mattshanevdberg** [comment]
dash_bootstrap_components not in the requirements file. Needs to be added

File: pages/EventTrail/components/event_trail_map.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+                [dl.TileLayer(), dl.LayerGroup(id=id)],
+                center=[-30, 25], zoom=5,
+                style={'width': '100%', 'height': '500px'}
+            )
+        ],
+        style={'zIndex': 10}
+    )
+
+def generate_markers(data):
+    # Create markers from data frame.
```
Comment (DylanGeldenhuys) [inline pages/EventTrail/components/event_trail_map.py:19]: Does this create the 10 day trail? There is not hovering on these markers that show time yet?

File: src/data_processor.py [hunk 2]
```diff
@@ -39,4 +39,9 @@ def generate_points(self):
         self.events['color'] = 'blue'
         self.events['size_normalised'] = (self.events['size'] - self.events['size'].min()) / (self.events['size'].max() - self.events['size'].min())
         return self
-        
\ No newline at end of file
+
+def fix_bad_coordinates(events):
```
Comment (DylanGeldenhuys) [inline src/data_processor.py:43]: Did we not have a function for this already?

File: src/db/queries.py [hunk 3]
```diff
@@ -151,10 +152,22 @@ def load_all_events(latitude, longitude, start_date, end_date, number_plates):
             queries.append(CameraEvent.created_at <= end_date)   
         if number_plates:
             queries.append(CameraEvent.number_plate.in_(number_plates))
-        result = session.query(CameraEvent.latitude, CameraEvent.longitude, CameraEvent.number_plate)\
+        result = session.query(CameraEvent.id, CameraEvent.latitude, CameraEvent.longitude, CameraEvent.number_plate)\
                 .filter(or_(CameraEvent.longitude == longitude, CameraEvent.longitude == latitude))\
                 .filter(or_(CameraEvent.latitude == latitude, CameraEvent.latitude == longitude))\
                 .filter(*queries)\
```
Comment (DylanGeldenhuys) [inline src/db/queries.py:158]: What does the * do here?

File: src/db/queries.py [hunk 4]
```diff
@@ -151,10 +152,22 @@ def load_all_events(latitude, longitude, start_date, end_date, number_plates):
...
                 .filter(or_(CameraEvent.longitude == longitude, CameraEvent.longitude == latitude))\
                 .filter(or_(CameraEvent.latitude == latitude, CameraEvent.latitude == longitude))\
                 .filter(*queries)\
                 .all()
 
-        return result
\ No newline at end of file
+        return result
+
+def load_event_trail_data(number_plate, date):
```
Comment (DylanGeldenhuys) [inline src/db/queries.py:163]: Simon I know you dont like duck typing, and you can actually specify the type in python with something like this

```
def greeting(name: str) -> str:
    return 'Hello ' + name
```

for future ref :)

Comment (DylanGeldenhuys) [reply src/db/queries.py:163]: That way the error will be throws if the type is wrong

**Pull request merged**